src/metabase/lib/fe_util.cljc

```diff
@@ -77,6 +77,24 @@
     (expand-temporal-expression expression-clause)
     expression-clause))
 
+(defn- expandable-case-or-if-expression?
+  [[op _options & _args]]
+  (or (= op :case)
+      (= op :if)))
+
+(defn- expand-case-or-if-expression
+  [[op options & args]]
+  (let [clauses (into [] (apply concat (first args)))
+        fallback (second args)
+        res (conj clauses fallback)]
+    (concat [op options] res)))
+
+(defn- maybe-expand-case-or-if-expression
+  [expression-clause]
+  (if (expandable-case-or-if-expression? expression-clause)
+    (expand-case-or-if-expression expression-clause)
+    expression-clause))
```

Comment:
```clojure
;; Discouraged
(if (p? x)
  (f x)
  x)
;; Encouraged
(cond-> x
  (p? x) f)
```
(See [Threading Macros](https://clojure.org/guides/threading_macros) if you want more details on `->` and friends; we use them a lot.)

```suggestion
  (cond-> expression-clause
    (expandable-case-or-if-expression? expression-clause) expand-case-or-if-expression))
```

Reply:
I managed to event remove the `maybe-` helpers altogether.